The test test_fifo_write_marker_does_no0 fails

```
============================================================================================== FAILURES ==============================================================================================
_______________________________________________________________________________ test_fifo_write_marker_does_not_block ________________________________________________________________________________
[gw0] freebsd15 -- Python 3.12.13 /usr/local/bin/python3.12

lock_file = '/tmp/pytest-of-yuri/pytest-487/popen-gw0/test_fifo_write_marker_does_no0/test.lock'

    def test_fifo_write_marker_does_not_block(lock_file: str) -> None:
        if sys.platform == "win32":
            pytest.skip("os.mkfifo is unix-only")
        marker = f"{lock_file}.write"
        os.mkfifo(marker)
        past = time.time() - 1000
        os.utime(marker, (past, past))
        # Without O_NONBLOCK this open blocks forever; the FIFO instead reads as a stale marker and is evicted.
        lock = _make_lock(lock_file)
        try:
>           with lock.write_lock(timeout=2):
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^

tests/soft_rw/test_soft_rw_sync.py:769: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/lib/python3.12/contextlib.py:137: in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
../stage/usr/local/lib/python3.12/site-packages/filelock/_soft_rw/_sync.py:269: in write_lock
    self.acquire_write(timeout, blocking=blocking)
../stage/usr/local/lib/python3.12/site-packages/filelock/_soft_rw/_sync.py:323: in acquire_write
    return self._acquire("write", timeout, blocking=blocking)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../stage/usr/local/lib/python3.12/site-packages/filelock/_soft_rw/_sync.py:486: in _acquire
    return self._do_acquire_inner(mode, timeout, start, blocking=blocking)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../stage/usr/local/lib/python3.12/site-packages/filelock/_soft_rw/_sync.py:504: in _do_acquire_inner
    marker_name, is_reader = self._acquire_writer_slot(token, deadline=deadline, blocking=blocking)
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../stage/usr/local/lib/python3.12/site-packages/filelock/_soft_rw/_sync.py:579: in _acquire_writer_slot
    self._wait_for(try_claim_writer, deadline=deadline, blocking=blocking)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <filelock._soft_rw._sync.SoftReadWriteLock object at 0x28d6b35d38f0>, predicate = <function SoftReadWriteLock._acquire_writer_slot.<locals>.try_claim_writer at 0x28d6b4355f80>

    def _wait_for(
        self,
        predicate: Callable[[], bool],
        *,
        deadline: float | None,
        blocking: bool,
    ) -> None:
        while True:
            if predicate():
                return
            now = time.perf_counter()
            if not blocking:
                raise Timeout(self.lock_file)
            if deadline is not None and now >= deadline:
>               raise Timeout(self.lock_file)
E               filelock._error.Timeout: The file lock '/tmp/pytest-of-yuri/pytest-487/popen-gw0/test_fifo_write_marker_does_no0/test.lock' could not be acquired.

../stage/usr/local/lib/python3.12/site-packages/filelock/_soft_rw/_sync.py:629: Timeout
========================================================================================== warnings summary ==========================================================================================
```

Version: 3.29.6
FreeBSD 15.1

--- a/src/filelock/_soft_rw/_sync.py
+++ b/src/filelock/_soft_rw/_sync.py
@@ -781,11 +781,16 @@ def _read_marker(name: str, *, dir_fd: int | None = None) -> tuple[_MarkerInfo |
     if fd is None:
         return None
     try:
-        try:
-            st = os.fstat(fd)
-            data = os.read(fd, _MAX_MARKER_SIZE + 1)
-        except OSError:  # pragma: no cover - e.g. EAGAIN from a hostile FIFO that has a writer attached
-            return None
+        st = os.fstat(fd)
+        # A legitimate marker is always a regular file, so anything else at the path -- above all a FIFO -- is
+        # reported as a malformed marker (its mtime still drives stale eviction) without being read. Reading is
+        # where platforms diverge: an empty non-blocking read yields 0 bytes on Linux/macOS but EAGAIN on
+        # FreeBSD, and the EAGAIN used to abort the stale-break and wedge the acquire until timeout (#587).
+        if not stat.S_ISREG(st.st_mode):
+            return None, st.st_mtime
+        data = os.read(fd, _MAX_MARKER_SIZE + 1)
+    except OSError:  # pragma: no cover - marker vanished or turned unreadable between open and read
+        return None
     finally:
         os.close(fd)
     return _parse_marker_bytes(data), st.st_mtime

--- a/tests/soft_rw/test_soft_rw_sync.py
+++ b/tests/soft_rw/test_soft_rw_sync.py
@@ -758,7 +758,7 @@ def test_stale_malformed_marker_is_evicted(lock_file: str, content: bytes) -> No
 
 def test_fifo_write_marker_does_not_block(lock_file: str) -> None:
     if sys.platform == "win32":
-        pytest.skip("os.mkfifo is unix-only")
+        pytest.skip("os.mkfifo is unix-only")  # also narrows sys.platform so ty resolves os.mkfifo below
     marker = f"{lock_file}.write"
     os.mkfifo(marker)
     past = time.time() - 1000
@@ -770,6 +770,28 @@ def test_fifo_write_marker_does_not_block(lock_file: str) -> None:
             pass
     finally:
         lock.close()
+
+
+def test_fifo_write_marker_with_writer_is_evicted(lock_file: str) -> None:
+    if sys.platform == "win32":
+        pytest.skip("os.mkfifo is unix-only")  # also narrows sys.platform so ty resolves os.mkfifo below
+    marker = f"{lock_file}.write"
+    os.mkfifo(marker)
+    past = time.time() - 1000
+    os.utime(marker, (past, past))
+    # A writer attached to the FIFO makes the non-blocking read raise EAGAIN on every platform, matching how a
+    # writerless FIFO already behaves on FreeBSD (#587). The stale marker must still be evicted by mtime, not
+    # read, so the acquire completes instead of timing out.
+    reader_fd = os.open(marker, os.O_RDONLY | os.O_NONBLOCK)
+    writer_fd = os.open(marker, os.O_WRONLY)
+    lock = _make_lock(lock_file)
+    try:
+        with lock.write_lock(timeout=2):
+            pass
+    finally:
+        lock.close()
+        os.close(writer_fd)
+        os.close(reader_fd)
 
 
 @pytest.mark.skipif(not hasattr(os, "O_NOFOLLOW"), reason="O_NOFOLLOW required")

